connectors/sources/sharepoint_online.py

```diff
@@ -764,27 +764,25 @@ async def site_pages(self, site_web_url):
         self._validate_sharepoint_rest_url(site_web_url)
 
         # select = "Id,Title,LayoutWebpartsContent,CanvasContent1,Description,Created,AuthorId,Modified,EditorId"
-        select = ""  # ^ is what we want, but site pages don't have consistent schemas, and this causes errors. Better to fetch all and slice
+        select = "*,EncodedAbsUrl"  # ^ is what we want, but site pages don't have consistent schemas, and this causes errors. Better to fetch all and slice
```

Comment:
TIL `*` is an option.

We're sure that `EncodedAbsUrl` is going to be available always, and not cause errors on some differently-schema'd pages?

Reply:
Looks like it's available for any list item object, so it's available for all Site Page objects because of it.